Add @automerge/automerge-repo-solid-primitives

chee opened the pull request:
yay. this adds a package like this: https://chee.github.io/solid-automerge/

alexjg reviewed (commented):
I'm not an expert in Solid but most stuff looks great here. Just a few CI files I think we don't need.

Are you okay to squash the history into a single commit with `Co-authord-by` trailers for the other contributors?

chee commented on packages/automerge-repo-solid-primitives/package.json:
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "@automerge/automerge-repo-solid-primitives",
+  "version": "2.0.7",
```
please alex stop releasing new automerges while i am working

chee commented:
> I'm not an expert in Solid but most stuff looks great here. Just a few CI files I think we don't need.
> 
> Are you okay to squash the history into a single commit 

done and done and done!

the CI is failing still because `vitest` fails rn if you run it from the root (if you `cd` into `./packages/automerge-repo-solid-primitives` and run them from there they will pass)

any ideas?

alexjg commented:
Ugh. Okay, I think we should go with having `pnpm test` use  `vitest run`. It would be nice if we could document somewhere exactly why we're doing that. Maybe update the section in `CONTRIBUTING.md`

chee commented:
> Ugh. Okay, I think we should go with having `pnpm test` use `vitest run`. It would be nice if we could document somewhere exactly why we're doing that. Maybe update the section in `CONTRIBUTING.md`

Okay, I've gone a different way, I've added [projects](https://vitest.dev/guide/projects) config Vitest root and then given each inner package a Vitest config that extends the root. I've added overrides for the ones that need it (WebSocket tests want to run on node, Solid tests need a strange setup to work in a Vitest workspace because of Solid's extremely annoying choice to be clever with export conditionals).

downside: more files.
upsides:
- all the tests pass
- all the tests run in the default vitest watcher mode
- it's colourful!`*`

> [!note]
> `* may be a downside depending on your sensibilities`

<img width="675" height="214" alt="" src="https://github.com/user-attachments/assets/da624d58-56f4-4304-a4a2-2479d6427118" />

alexjg commented:
Alright that is lovely
